src/System.Net.Sockets/src/System/Net/Sockets/TCPClient.cs

```diff
@@ -181,10 +199,22 @@ public void Connect(string hostname, int port)
                             Debug.Assert(address.AddressFamily == AddressFamily.InterNetwork || address.AddressFamily == AddressFamily.InterNetworkV6);
                             if ((address.AddressFamily == AddressFamily.InterNetwork && Socket.OSSupportsIPv4) || Socket.OSSupportsIPv6)
                             {
-                                tmpSocket = new Socket(address.AddressFamily, SocketType.Stream, ProtocolType.Tcp);
-                                tmpSocket.Connect(address, port);
-                                _clientSocket = tmpSocket;
-                                tmpSocket = null;
+                                var socket = new Socket(address.AddressFamily, SocketType.Stream, ProtocolType.Tcp);
+                                Volatile.Write(ref _clientSocket, socket);
+                                if (Volatile.Read(ref _cleanedUp) == 1)
+                                {
+                                    socket.Dispose();
```

Comment:
We dispose of the socket and then immediately try to Connect on it?

Reply:
To make this throw ObjectDisposedException.